apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/TraceSnippets.cs

```diff
@@ -0,0 +1,269 @@
+            Assert.False(response.Headers.Contains(TraceHeaderContext.TraceHeader));
+        }
+
+        public void Dispose()
+        {
+            _client.Dispose();
+            _server.Dispose();
+        }
+    }
+
+    /// <summary>
+    /// A simple web application to test the <see cref="CloudTraceMiddleware"/>
+    /// and associated classes.
+    /// </summary>
+    internal class TraceTestApplication
+    {
+        private static readonly string ProjectId = Utils.GetProjectIdFromEnvironment();
+
+        // Sample: RegisterGoogleTracer
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // The line below is needed for trace ids to be added to logs.
+            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+
+            // Replace ProjectId with your Google Cloud Project ID.
+            services.AddGoogleTrace(options =>
+            {
+                options.ProjectId = ProjectId;
+            });
+
+            services.AddMvc();
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            // Use at the start of the request pipeline to ensure the entire request is traced.
+            app.UseGoogleTrace();
+
+            // ...any other configuration your application requires.
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+        // End sample
+    }
+
+    /// <summary>
+    /// A controller for the <see cref="TraceTestApplication"/> used to trace operations.
+    /// </summary>
+    public class TraceSamplesController : Controller
+    {
+        // Sample: UseTracer
+        /// <summary>
+        /// Manually trace a set of operations.
+        /// The <see cref="IManagedTracer"/> is populated by dependency injection
+        /// thanks to the use of the <see cref="FromServicesAttribute"/> attribute.
+        /// </summary>
+        public void TraceHelloWorld(string id, [FromServices] IManagedTracer tracer)
+        {
+            // Change the name of the span to what makese sense in your context.
+            using (tracer.StartSpan(id))
+            {
+                // The code whose execution is to be included in the span goes here.
+                ViewData["Message"] = "Hello World.";
+            }
+        }
+        // End sample
+
+        // Sample: UseTracerRunIn
+        /// <summary>
+        /// Manually trace a specific Action or Func<T>.
+        /// The <see cref="IManagedTracer"/> is populated by dependency injection
+        /// thanks to the use of the <see cref="FromServicesAttribute"/> attribute.
+        /// </summary>
+        public void TraceHelloWorldRunIn(string id, [FromServices]IManagedTracer tracer)
+        {
+            tracer.RunInSpan(
+                // The Action or Func<T> to be traced.
+                () =>
```

Comment:
Can we compress the formatting here?  Maybe move this up to the RunInSpan Line?

Reply:
I did like that to add the comments clearly. I'm not sure all those comments are necessary in the snippets, if you think they are not, I can compress the formatting, just confirm.